src/components/Collection/CollectionList.tsx

```diff
@@ -0,0 +1,100 @@
+import { useAuthenticator } from '@aws-amplify/ui-react';
```

Comment:
I would recommend separating your API handlers from your component definitions. Try and make your components as generic as possible and just have them get their data via parameters or call this separate functions in useEffect